From e72c1a01c7ffc614fb6b6d570571e09828bb8cc2 Mon Sep 17 00:00:00 2001
From: Alyson Deives Pereira <alyson.deivespereira@windriver.com>
Date: Wed, 22 Apr 2026 08:16:32 -0300
Subject: [PATCH] csi-powermax: add secrets in templates

This patch consists of adding a secrets in the helm chart templates
directory. It's needed because this helm chart needs a config secrets
that should be created manually.

Signed-off-by: Alyson Deives Pereira <alyson.deivespereira@windriver.com>
---
 .../templates/default-secret.yaml             | 18 ++++++++++++++++
 charts/csi-powermax/templates/secrets.yaml    | 21 +++++++++++++++++++
 charts/csi-powermax/values.yaml               |  6 +++++-
 3 files changed, 44 insertions(+), 1 deletion(-)
 create mode 100644 charts/csi-powermax/templates/default-secret.yaml
 create mode 100644 charts/csi-powermax/templates/secrets.yaml

diff --git a/charts/csi-powermax/templates/default-secret.yaml b/charts/csi-powermax/templates/default-secret.yaml
new file mode 100644
index 0000000..8f80677
--- /dev/null
+++ b/charts/csi-powermax/templates/default-secret.yaml
@@ -0,0 +1,18 @@
+{{- if .Values.global.defaultSecret }}
+apiVersion: v1
+kind: Secret
+metadata:
+{{- if .Values.global.defaultCredentialsSecret }}
+  name: {{ .Values.global.defaultCredentialsSecret }}
+{{- else }}
+  name: {{ .Release.Name }}-creds
+{{- end }}
+  namespace: {{ .Release.Namespace }}
+type: Opaque
+stringData:
+  username: {{ .Values.global.defaultSecret.username }}
+  password: {{ .Values.global.defaultSecret.password }}
+  {{- if and .Values.enableCHAP .Values.global.defaultSecret.chapsecret }}
+  chapsecret: {{ .Values.secret.chapsecret }}
+  {{- end }}
+{{- end }}
diff --git a/charts/csi-powermax/templates/secrets.yaml b/charts/csi-powermax/templates/secrets.yaml
new file mode 100644
index 0000000..65af38b
--- /dev/null
+++ b/charts/csi-powermax/templates/secrets.yaml
@@ -0,0 +1,21 @@
+{{- range $i, $ms := $.Values.global.managementServers }}
+{{- if $ms.credentialsSecret }}
+apiVersion: v1
+kind: Secret
+metadata:
+{{- if $ms.credentialsSecret }}
+  name: {{ $ms.credentialsSecret }}
+{{- else }}
+  name: {{ $.Release.Name }}-creds-{{ $i }}
+{{- end }}
+  namespace: {{ $.Release.Namespace }}
+type: Opaque
+stringData:
+  username: {{ $ms.username }}
+  password: {{ $ms.password }}
+  {{- if and $.Values.enableCHAP $ms.chapsecret }}
+  chapsecret: {{ $ms.chapsecret }}
+  {{- end }}
+---
+{{- end }}
+{{- end }}
diff --git a/charts/csi-powermax/values.yaml b/charts/csi-powermax/values.yaml
index a2e2e8d..2d67a74 100644
--- a/charts/csi-powermax/values.yaml
+++ b/charts/csi-powermax/values.yaml
@@ -24,7 +24,11 @@ global:
   # The name of the Kubernetes Secret containing the details of the PowerMax arrays,
   # their Unisphere endpoints and their login credentials.
   # Default: "powermax-creds"
-  defaultCredentialsSecret: "powermax-creds"
+  defaultCredentialsSecret: csi-powermax-creds
+  defaultSecret:
+    username:
+    password:
+    chapsecret:
   # useSecret
   # Defines if a Secret should be used to provide Unisphere for PowerMax endpoints
   # and login credentials instead of the deprecated powermax-reverseproxy-config ConfigMap.
--
2.34.1
